frontend/src/app/(routes)/staking/components/DialogAllValidators.tsx

```diff
@@ -0,0 +1,487 @@
+
+  useEffect(() => {
+    if (validators?.activeSorted.length < PER_PAGE) {
+      setSlicedValidators(validators?.activeSorted);
+    } else {
+      setCurrentPage(1);
+      setSlicedValidators(validators?.activeSorted?.slice(0, 1 * PER_PAGE));
+    }
+  }, [validators?.activeSorted]);
+
+  useEffect(() => {
+    const filteredValidators = validators?.activeSorted.filter(
+      (validator) =>
+        validators.active[validator]?.description.moniker
+          .toLowerCase()
+          .includes(searchTerm.toLowerCase())
+    );
+    setFiltered(filteredValidators);
+    setCurrentPage(1);
+  }, [searchTerm, validators?.activeSorted]);
+  return (
+    <>
+      <div className="px-10">
+        {searchTerm.length ? (
+          <>
+            <Filtered
+              filtered={filtered}
+              validators={validators}
+              active={true}
+            />
+          </>
+        ) : (
+          <>
+            <div className="flex flex-col gap-6">
+              {slicedValidators?.map((validator, index) => {
+                const moniker =
+                  validators.active[validator]?.description.moniker;
+                const identity =
+                  validators.active[validator]?.description.identity;
+                const commission =
+                  Number(
+                    validators.active[validator]?.commission?.commission_rates
+                      .rate
+                  ) * 100;
+                const jailed = validators.active[validator]?.jailed;
+                const status = validators.active[validator]?.status;
+                const rank = getValidatorRank(
+                  validator,
+                  validators.activeSorted
+                );
+
+                return (
+                  <ValidatorComponent
+                    key={index + PER_PAGE * (currentPage - 1)}
+                    moniker={moniker}
+                    identity={identity}
+                    commission={commission}
+                    jailed={jailed}
+                    status={status}
+                    active={true}
+                    rank={rank}
+                  />
+                );
+              })}
+            </div>
+            <div className="w-full h-[0.25px] bg-[#FFFFFF66] my-6"></div>
+            <div className="absolute bottom-12 right-10">
+              <Pagination
+                sx={paginationComponentStyles}
+                count={Math.ceil(validators?.activeSorted?.length / PER_PAGE)}
+                shape="circular"
+                onChange={(_, v) => {
+                  setCurrentPage(v);
+                  setSlicedValidators(
+                    validators?.activeSorted?.slice(
+                      (v - 1) * PER_PAGE,
+                      v * PER_PAGE
+                    )
+                  );
+                }}
+              />
+            </div>
+          </>
+        )}
+      </div>
+    </>
+  );
+};
+
+const InactiveValidators = ({
```

Comment:
Based on my understanding, you are currently creating a single file for dialogue and incorporating its components in the same file. However, if we plan on adding more components in the future, the file size could become too large. It may be better to write each independent component in a separate file to avoid this issue.